test/unit/streaming.ts

```diff
@@ -257,6 +257,71 @@ describe('streaming', () => {
     }, 50);
   });
 
+  it('Not receive metadata event', done => {
```

Comment:
Do we have a test coverage to check the `responseHasSent` logic? Probably add a counter to the test and check that the response event was emitted just once.

Reply:
Add two test cases to cover the `responseHasSent`:
1. Stream received both `metadata` and `status` events. 
- `response` should contains `metadata` field.
2. Stream received only `status` events.
- `response` should has no `metadata` field.